File: WTCChatApp/Models/Notification.swift
//
//  Notification.swift
//  WTCChatApp
//
//  Created by WTC Challenge
//

import Foundation

struct AppNotification: Codable, Identifiable {
    let id: UUID
    var userId: UUID
    var title: String
    var body: String
    var type: String
    var read: Bool
    var messageId: UUID?
    var createdAt: Date

    enum CodingKeys: String, CodingKey {
        case id
        case userId = "user_id"
        case title
        case body
        case type
        case read
        case messageId = "message_id"
        case createdAt = "created_at"
    }

    init(id: UUID = UUID(), userId: UUID, title: String, body: String,
         type: String = "message", read: Bool = false, messageId: UUID? = nil,
         createdAt: Date = Date()) {
        self.id = id
        self.userId = userId
        self.title = title
        self.body = body
        self.type = type
        self.read = read
        self.messageId = messageId
        self.createdAt = createdAt
    }
}

File: backend/src/main/java/com/wtc/chatapp/model/Notification.java
package com.wtc.chatapp.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.index.Indexed;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.Instant;
import java.util.UUID;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Document(collection = "notifications")
public class Notification {

    @Id
    @Builder.Default
    private String id = UUID.randomUUID().toString();

    @Indexed
    private String userId;

    private String title;

    private String body;

    @Builder.Default
    private NotificationType type = NotificationType.MESSAGE;

    @Builder.Default
    private boolean read = false;

    private String messageId;

    @Builder.Default
    private Instant createdAt = Instant.now();
}
